src/datanode/src/heartbeat.rs

```diff
@@ -83,8 +102,13 @@ impl HeartbeatTask {
         Ok(tx)
     }
 
-    async fn handle_response(resp: HeartbeatResponse) {
+    async fn handle_response(
+        ctx: HeartbeatResponseHandlerContext,
+        resp: HeartbeatResponse,
+        handler_executor: HeartbeatResponseHeadlerExecutorRef,
+    ) -> Result<()> {
         trace!("heartbeat response: {:?}", resp);
+        handler_executor.handle(ctx, resp).await
```

Comment:
Some time-costing operations will spawn a background thread for processing. However, these responses might be replied to MetaSrv by the next heartbeat.